### ISSUE TITLE
k3d cluster creation fails in codespaces

### ISSUE THREAD

**Issue by Reshrahim** [radius-project/samples#689]
User facing docs - https://docs.radapp.io/getting-started/

There seems to be an issue with k3d cluster creation when we spin up a devcontainer via codespaces. 
Error from the creation log 
 
```
2023-10-24 13:40:27.779Z: ERRO[0001] Failed to get nodes for cluster 'k3s-default': docker failed to get containers with labels 'map[k3d.cluster:k3s-default]': failed to list containers: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
2023-10-24 13:40:27.816Z: INFO[0000] portmapping '8081:80' targets the loadbalancer: defaulting to [servers:*:proxy agents:*:proxy]
2023-10-24 13:40:27.817Z: ERRO[0000] Failed to get nodes for cluster 'k3s-default': docker failed to get containers with labels 'map[k3d.cluster:k3s-default]': failed to list containers: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
2023-10-24 13:40:27.817Z: INFO[0000] Prep: Network                                
2023-10-24 13:40:27.818Z: ERRO[0000] Failed Cluster Preparation: Failed Network Preparation: failed to create cluster network: failed to check for duplicate docker networks: docker failed to list networks: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
ERRO[0000] Failed to create cluster >>> Rolling Back
 ```
when I ran the cluster create command manually in the codespaces, the cluster got created succesfully. Not sure whats happening in the post create script. 

May be a permission issue or a readiness issue? 

[AB#9902](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/9902)

**Comment by Reshrahim** [comment]
@jasonviviano is currently investigating to see if there is a race condition

**Comment by willtsai** [comment]
note that I've commented out the codespaces instructions in this page, so we need to add it back when this issue gets fixed: https://github.com/radius-project/docs/pull/864

**Comment by jasonviviano** [comment]
Closing, fixed with: #694
